torch/_torch_docs.py

```diff
@@ -1695,18 +1695,25 @@ def merge_dicts(*dicts):
            r"""
 .. function:: div(input, other, out=None) -> Tensor
 
-Divides each element of the input :attr:`input` with the scalar :attr:`other`
-and returns a new resulting tensor.
+Divides each element of the input ``input`` with the scalar ``other`` and
+returns a new resulting tensor.
 
 .. math::
-    \text{out}_i = \frac{\text{input}_i}{\text{other}}
+    \text{{out}}_i = \frac{{\text{{input}}_i}}{{\text{{other}}}}
 
-If :attr:`input` is of type `FloatTensor` or `DoubleTensor`, :attr:`other`
-should be a real number, otherwise it should be an integer
+If the :class:`torch.dtype` of ``input`` and ``other`` differ, the
+:class:`torch.dtype` of the result tensor is determined following rules
+described in the type promotion :ref:`documentation <type-promotion-doc>`. If
+``out`` is specified, the result must be :ref:`castable <type-promotion-doc>`
+to the :class:`torch.dtype` of the specified output tensor. If ``input`` is
+integral, and ``other`` is integral and zero, the behavior is undefined and
+arbitrary.
```

Comment:
One thing that could be slightly unclear about this is that in this context False is considered integral and 0.
e.g. `torch.tensor(1)/False` is a segfault.

Reply:
instead of trying to spell out types accurately for each attribute, I might prefer keeping it simple and just saying something like "Integral division by zero leads to undefined behavior."

Reply:
Updated!